### PR TITLE
Added serialization support to the tooling API

### PR DESCRIPTION
**Pull request by jsight**

### CODE REVIEW

**Comment by jsight** [comment]
retest

**Comment by jsight** [comment]
retest

File: tooling/impl/src/main/java/org/jboss/windup/tooling/ExecutionResultsImpl.java [hunk 1]
```diff
@@ -33,37 +41,67 @@
  * @author <a href="mailto:jesse.sightler@gmail.com">Jesse Sightler</a>
  * @author <a href="mailto:hotmana76@gmail.com">Marek Novotny</a>
  */
+@XmlRootElement(name = "execution-results")
 public class ExecutionResultsImpl implements ExecutionResults
 {
```
Comment (OndraZizka) [inline tooling/impl/src/main/java/org/jboss/windup/tooling/ExecutionResultsImpl.java]: Isn't this class mixing data and logic too much? Maybe methods working with GraphContext could be somewhere else or at least static?

**Comment by OndraZizka** [comment]
Leaving to John to decide

File: tooling/impl/src/main/java/org/jboss/windup/tooling/ExecutionResultsImpl.java (hunk 1 above)
Comment (jsight) [inline tooling/impl/src/main/java/org/jboss/windup/tooling/ExecutionResultsImpl.java]: Yes, that is probably true. It could probably use some refactoring.

**Pull request merged**